// commands.proto
// build when new commands are added
// with nano pb:
// python C:\DevTools\nanopb\generator\nanopb_generator.py .\src\core\proto\ClusterMessagesProto.proto
// THen I modify the .dc file so the .h include is filebame only
// in .h, after #include <pb.h>, I add the following:
/*
#include "pb_encode.h"
#include "pb_decode.h"
#include "pb_common.h"
*/
//
// in ClusterMessagesProto.pb.c I change:
// #include "src/core/proto/ClusterMessagesProto.pb.h"
// to:
// #include "ClusterMessagesProto.pb.h"
syntax = "proto3";

enum CommandTypeProto {
    FILL_NODE = 0;
    BLIT_NODE = 1;
    FILL_BUFFER = 2;
    BLIT_BUFFER = 3;
    SET_NODE_PIXEL = 4;
    QUEUE_NODE_PIXEL = 5;
    DEQUEUE_NODE_PIXEL = 6;
}

message ClusterMessageProto {
    int32 cluster_id = 1;
    CommandTypeProto command_type = 2;

    int32 param_node_id = 3;
    int32 param_pixel_index = 4;
    uint32 param_color = 5;
    uint32 param_pad_color = 6;
    repeated uint32 param_colors = 7;
}
